### ISSUE TITLE
Checking host firewall: OK but Client complains about ports

### ISSUE THREAD

**Issue by mdibaiee** [Jigsaw-Code/outline-server#217]
Hi,

I installed Outline Server on my VPS about a month ago and I fired up the manager, created a few keys, and shared them. Everything went well and was great.

A few days ago I opened the manager again, and to my surprise it wouldn't connect to the server. I tried re-installing the server and trying with a new `apiUrl` and `certSha256`, but I keep getting

> Your Outline Server was installed correctly, but we are not able to connect to it. Most likely this is because your server's firewall rules are blocking incoming connections. Please review them and make sure to allow incoming TCP connections on ports ranging from 1024 to 65535.

Meanwhile, re-instaling the server passes the `Checking host firewall`, and it might help to note that my previously shared keys still work well, it's just I can't connect the manager to add new keys.

Manager version: `1.1.4`
Installed using this script:
```
https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh
```

**Comment by mdibaiee** [comment]
By the way, I am using DigitalOcean droplets (but I used the advanced setup because I am using an already-setup VPS), and I checked the firewall settings in the panel and the external firewall is not enabled.

**Comment by trevj** [comment]
@mdibaiee Thanks for trying Outline. Apologies for the slow reply.

In your iptables output I see several input rules. I'm no ufw expert but if ufw were disabled I would expect to see fewer rules. Can you double-check that you ran?:
```
ufw disable
```

**Comment by mdibaiee** [comment]
@trevj Hey, I forgot to mention, but I have `ufw` disabled as well.

So

```
$ sudo ufw status
Status: inactive
```

**Comment by trevj** [comment]
@mdibaiee Yes but there's more rules in the `INPUT` table than I would expect, given ufw is disabled. Have you tried:
* in the Outline server manager, "forget" the server
* on your VPS, re-run the install script
* re-add your server to the Outline server manager, by pasting the output of the script

?

**Comment by trevj** [comment]
@mdibaiee Yes but there's more rules in the `INPUT` table than I would expect, given ufw is disabled. Have you tried:
* in the Outline server manager, "forget" the server
* on your VPS, re-run the install script
* re-add your server to the Outline server manager, by pasting the output of the script

?

**Comment by mdibaiee** [comment]
@trevj 
1. The Outline Manager has already *forgot* my server, so when I open it up it asks me to set up the server and input the API credentials
2. I have re-run the install script tens of times, using different methods:
  * Simply re-running on a past installation
  * Deleting all docker images and active containers and re-running the script
  * Clearing the iptable, deleting docker images and running the installation again
3. Afterward, I would paste the new credentials, but still the same output, the install script reports firewall as being "OK" but the manager complains about something being wrong with it.

All the while, a client using a key generated using an old installation on the same server works fine (Android Outline Client)

**Comment by lgg** [comment]
Same issue. New server, can't connect. Another Outline Server(installed 4 months ago works fine).

**Comment by rmeuth01** [comment]
I'm having a similar problem.  I accidentally deleted my server on google compute, tried to reinstall today using the command line tools.  I found I had to install docker and add it to my user group manually before I ran the script.   I opened the firewall ports as instructed, firewall check says "OK", but outline manager says "Your Outline Server was installed correctly, but we are not able to connect to it. "  I rolled back to a previous version of the outline manager (downloaded 7/23) and it worked.

**Comment by DonaldKBrown** [comment]
I'm having a similar issue, as well. I am currently running UFW, but I've tried with and without. I've done `ufw allow 1024:65535/tcp` and `ufw allow 1024:65535/udp` as well as disabling UFW and even going as far as just setting my iptables completely open and accepting on all ports.

I am able to telnet into the server management port, but I cannot connect to any other port used by any of my keys. This is running on a dedicated server (not VPS) through Online.net in the Netherlands with Ubuntu 18.04 installed.

**Comment by 7urkm3n** [comment]
if anyone has the firewall issue, just flush the rules `sudo iptables -F` and try again. Make sure docker restart or better server restart `sudo reboot`. In my case reinstalled the outline-server, deleting `docker stop $(docker ps -aq)` and `docker rm $(docker ps -aq)` and `docker rmi $(docker images -q)`.

**Comment by mdibaiee** [comment]
@7urkm3n Thank you, I just went through all of the commands you mentioned and it seems to have solved the issue, or maybe it's the new manager that's working.

**Comment by 7urkm3n** [comment]
> @7urkm3n Thank you, I just went through all of the commands you mentioned and it seems to have solved the issue, or maybe it's the new manager that's working.

Glad it helped. I was installing it last night on VPS and Dedicated servers, both had an issue with firewall. Just flushing the rules helped me more than overriding some rules. Later, you can update rules for other services also.

**Comment by sriram89194** [comment]
> Hi,
> 
> I installed Outline Server on my VPS about a month ago and I fired up the manager, created a few keys, and shared them. Everything went well and was great.
> 
> A few days ago I opened the manager again, and to my surprise it wouldn't connect to the server. I tried re-installing the server and trying with a new `apiUrl` and `certSha256`, but I keep getting
> 
> > Your Outline Server was installed correctly, but we are not able to connect to it. Most likely this is because your server's firewall rules are blocking incoming connections. Please review them and make sure to allow incoming TCP connections on ports ranging from 1024 to 65535.
> 
> Meanwhile, re-instaling the server passes the `Checking host firewall`, and it might help to note that my previously shared keys still work well, it's just I can't connect the manager to add new keys.
> 
> Manager version: `1.1.4`
> Installed using this script:
> 
> ```
> https://raw.githubusercontent.com/Jigsaw-Code/outline-server/master/src/server_manager/install_scripts/install_server.sh
> ```
Your Outline Server was installed correctly, but we are not able to connect to it. Most likely this is because your server's firewall rules are blocking incoming connections. Please review them and make sure to allow incoming TCP connections on ports ranging from 1024 to 65535.
----->I'm facing this issue that what I mentioned above. While I'm trying to setup the server using outline manager from the goole cloud platform.
![Screenshot (63)](https://user-images.githubusercontent.com/54448841/81772271-118b3e00-9503-11ea-8558-4bc2f31b24f3.png)
----->So, please help me with this issue.And let me know what to do.
Thank You.

**Comment by hbsoftco** [comment]
@sriram89194 
Hi, I have this problem too. How did you solve the problem?

**Comment by seyeeet** [comment]
also have this issue on ubuntu

**Comment by sriram89194** [comment]
> 

I tried reinstalling it, and then it worked

**Comment by NickSdot** [comment]
Had the same problem on Ubuntu. Output after the installation mentioned this:

> Make sure to open the following ports on your firewall, router or cloud provider:
> -- Management port 111111, for TCP
> -- Access key port 222222, for TCP and UDP

Opened the missing ports via `ufw`

**Commands & output**
```
root@myserver:~# ufw allow proto tcp to any port 111111
Rule added
Rule added (v6)
root@myserver:~# ufw allow proto udp to any port 222222
Rule added
Rule added (v6)
root@myserver:~# ufw allow proto tcp to any port 222222
Rule added
Rule added (v6)
```

**Check & expected result:**
```
root@mein1vpnserver:~# ufw status
Status: active

To                         Action      From
--                         ------      ----
22/tcp                     ALLOW       Anywhere
111111/tcp                  ALLOW       Anywhere
222222/udp                  ALLOW       Anywhere
222222/tcp                  ALLOW       Anywhere
22/tcp (v6)                ALLOW       Anywhere (v6)
111111/tcp (v6)             ALLOW       Anywhere (v6)
222222/udp (v6)             ALLOW       Anywhere (v6)
222222/tcp (v6)             ALLOW       Anywhere (v6)
```

**Comment by ITGeekCoder** [comment]
Sorry! But do I have to post a new issue or will my earlier post be seen in the future!?

**Comment by maddyhof** [comment]
Thanks for opening a new issue #1357. Closing this one.
